Incorporate unet

navjot-nangia opened the pull request:
updated metadata schema; fixed bbox_xywh_dict

navjot-nangia committed:
updated .gitignore

navjot-nangia committed:
changed directory for main scripts; added paths in config folder and made changes in scripts for it; added unet_segment_weeds script along with it's utils

navjot-nangia committed:
update pipeline configuration to use unet_segment_weeds for inference

navjot-nangia committed:
updated config structure; updated scripts to go along the updated; incorporated unet into this pipeline

navjot-nangia committed:
rearranged for better logging and readability

navjot-nangia committed:
removed 'collection_timing' and 'collection_location' from the json

navjot-nangia committed:
added metadate schema

navjot-nangia committed:
added metadata version into metadata

navjot-nangia committed:
changed config for metadata

navjot-nangia committed:
updated metadata schema; corrected the bbox_xywh_dict

mkutu commented:
holding off on merging until changes to species_info.json have been finalized

navjot-nangia committed:
edited unet script to account for the width height in bbox

navjot-nangia commented:
fixed unet.py to account for changed bbox.

navjot-nangia committed:
added species_info.json. it will be eventually moved to other location.

navjot-nangia committed:
renamed directory for developed- images to use hyphen instead of underscore. This is how the directory is named in the downloaded batches.

mkutu commented on species_info.json:
use empty lists instead of null. We plan to iterate through alias and can avoid any additional null checks. An empty list ([]) is preferable. It keeps data structures consistent and avoids potential null-related errors.

```json
{
    "species": {
        "background": {
            "class_id": 0,
            "USDA_symbol": "background",
            "EPPO": "background",
            "group": "background",
            "class": "background",
            "subclass": "background",
            "order": "background",
            "family": "background",
            "genus": "background",
            "species": "background",
            "common_name": "background",
            "authority": "background",
            "growth_habit": "background",
            "duration": "background",
            "collection_location": "background",
            "category": "background",
            "collection_timing": "background",
            "multi_species_USDA_symbol": null,
            "link": null,
            "note": null,
            "hex": "#000000",
            "rgb": [
                0,
                0,
                0
            ],
            "alias": []
        },
        "AMPA": {
            "class_id": 1,
            "USDA_symbol": "AMPA",
            "EPPO": "AMAPA",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Caryophyllidae",
            "order": "Caryophyllales",
            "family": "Amaranthaceae",
            "genus": "Amaranthus",
            "species": "palmeri",
            "common_name": "Palmer amaranth",
            "authority": "Watson",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=AMPA",
            "note": null,
            "hex": "#1d686e",
            "rgb": [
                29,
                104,
                110
            ],
            "alias": []
        },
        "AMAR2": {
            "class_id": 2,
            "USDA_symbol": "AMAR2",
            "EPPO": "AMBEL",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Asterales",
            "family": "Asteraceae",
            "genus": "Ambrosia",
            "species": "artemisiifolia",
            "common_name": "Common ragweed",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=AMAR2",
            "note": null,
            "hex": "#e452f1",
            "rgb": [
                228,
                82,
                241
            ],
            "alias": []
        },
        "SEOB4": {
            "class_id": 3,
            "USDA_symbol": "SEOB4",
            "EPPO": "CASOB",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Rosidae",
            "order": "Fabales",
            "family": "Fabaceae",
            "genus": "Senna",
            "species": "obtusifolia",
            "common_name": "Sicklepod",
            "authority": "(Linnaeus) Irwin and Barneby",
            "growth_habit": "forb/herb, subshrub",
            "duration": "annual perenial",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=SEOB4",
            "note": null,
            "hex": "#60e54e",
            "rgb": [
                96,
                229,
                78
            ],
            "alias": []
        },
        "XAST": {
            "class_id": 4,
            "USDA_symbol": "XAST",
            "EPPO": "XANST",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Asterales",
            "family": "Asteraceae",
            "genus": "Xanthium",
            "species": "strumarium",
            "common_name": "Cocklebur",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=XAST",
            "note": null,
            "hex": "#903be2",
            "rgb": [
                144,
                59,
                226
            ],
            "alias": [
                "common cocklebur"
            ]
        },
        "DISA": {
            "class_id": 5,
            "USDA_symbol": "DISA",
            "EPPO": "DIGSA",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Digitaria",
            "species": "sanguinalis",
            "common_name": "Large crabgrass",
            "authority": "(Linnaeus) Scopoli",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=DISA",
            "note": "Large crabgrass may be southern crabgrass",
            "hex": "#9ee33f",
            "rgb": [
                158,
                227,
                63
            ],
            "alias": [
                "crabgrass (large or other spp.)"
            ]
        },
        "ELIN3": {
            "class_id": 6,
            "USDA_symbol": "ELIN3",
            "EPPO": "ELEIN",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Eleusine",
            "species": "indica",
            "common_name": "Goosegrass",
            "authority": "(Linnaeus) G\u00e4rtner",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=ELIN3",
            "note": null,
            "hex": "#5d4ce7",
            "rgb": [
                93,
                76,
                231
            ],
            "alias": []
        },
        "URPL2": {
            "class_id": 7,
            "USDA_symbol": "URPL2",
            "EPPO": "BRAPP",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Urochloa",
            "species": "platyphylla",
            "common_name": "Broadleaf signalgrass",
            "authority": "(Munro ex C. Wright) R.D. Webster",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=URPL2",
            "note": null,
            "hex": "#dbe93c",
            "rgb": [
                219,
                233,
                60
            ],
            "alias": []
        },
        "CYRO": {
            "class_id": 8,
            "USDA_symbol": "CYRO",
            "EPPO": "CYPRO",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Cyperaceae",
            "genus": "Cyperus",
            "species": "rotundus",
            "common_name": "Purple nutsedge",
            "authority": "Linnaeus",
            "growth_habit": "graminoid",
            "duration": "perennial",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=CYRO",
            "note": "Possibly yellow nutsedge as well",
            "hex": "#bd36d6",
            "rgb": [
                189,
                54,
                214
            ],
            "alias": []
        },
        "AMTU": {
            "class_id": 9,
            "USDA_symbol": "AMTU",
            "EPPO": "AMATU",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Caryophyllidae",
            "order": "Caryophyllales",
            "family": "Amaranthaceae",
            "genus": "Amaranthus",
            "species": "tuberculatus",
            "common_name": "Waterhemp",
            "authority": "(Moquin-Tandon) Sauer",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "TX",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=AMTU",
            "note": null,
            "hex": "#3fac38",
            "rgb": [
                63,
                172,
                56
            ],
            "alias": [
                "common waterhemp"
            ]
        },
        "ECCR": {
            "class_id": 10,
            "USDA_symbol": "ECCR",
            "EPPO": "ECHCG",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Echinochloa",
            "species": "crus-galli",
            "common_name": "Barnyardgrass",
            "authority": "(Linnaeus) Palisot de Beauvois ",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "TX",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=ECCR",
            "note": null,
            "hex": "#912bba",
            "rgb": [
                145,
                43,
                186
            ],
            "alias": []
        },
        "ECCO2": {
            "class_id": 11,
            "USDA_symbol": "ECCO2",
            "EPPO": "ECHCO",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Echinochloa",
            "species": "colona",
            "common_name": "Jungle rice",
            "authority": "(Linnaeus) Link",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "TX",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=ECCO2",
            "note": null,
            "hex": "#57eb93",
            "rgb": [
                87,
                235,
                147
            ],
            "alias": [
                "junglerice"
            ]
        },
        "URTE2": {
            "class_id": 12,
            "USDA_symbol": "URTE2",
            "EPPO": "PANTE",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Urochloa",
            "species": "texana",
            "common_name": "Texas millet",
            "authority": "(Buckley) Webster",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=URTE2",
            "note": null,
            "hex": "#e342c6",
            "rgb": [
                227,
                66,
                198
            ],
            "alias": []
        },
        "BASC5": {
            "class_id": 13,
            "USDA_symbol": "BASC5",
            "EPPO": "KCHSC",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Caryophyllidae",
            "order": "Caryophyllales",
            "family": "Chenopodiaceae",
            "genus": "Bassia",
            "species": "scoparia",
            "common_name": "Kochia",
            "authority": "(Linnaeus) Scott",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "TX",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=BASC5",
            "note": null,
            "hex": "#8fe175",
            "rgb": [
                143,
                225,
                117
            ],
            "alias": []
        },
        "HEAN3": {
            "class_id": 14,
            "USDA_symbol": "HEAN3",
            "EPPO": "HELAN",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Asterales",
            "family": "Asteraceae",
            "genus": "Helianthus",
            "species": "annuus",
            "common_name": "Common sunflower",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "TX",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=HEAN3",
            "note": null,
            "hex": "#5a42c1",
            "rgb": [
                90,
                66,
                193
            ],
            "alias": []
        },
        "PAHY": {
            "class_id": 15,
            "USDA_symbol": "PAHY",
            "EPPO": "PTNHY",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Asterales",
            "family": "Asteraceae",
            "genus": "Parthenium",
            "species": "hysterophorus",
            "common_name": "Ragweed parthenium",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "TX",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=PAHY",
            "note": null,
            "hex": "#e3ca3b",
            "rgb": [
                227,
                202,
                59
            ],
            "alias": []
        },
        "SOHA": {
            "class_id": 16,
            "USDA_symbol": "SOHA",
            "EPPO": "SORHA",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Sorghum",
            "species": "halepense",
            "common_name": "Johnsongrass",
            "authority": "(Linnaeus) Persoon",
            "growth_habit": "graminoid",
            "duration": "perennial",
            "collection_location": "TX",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=SOHA",
            "note": null,
            "hex": "#6364e2",
            "rgb": [
                99,
                100,
                226
            ],
            "alias": []
        },
        "GLMA4": {
            "class_id": 17,
            "USDA_symbol": "GLMA4",
            "EPPO": " GLXMA",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Rosidae",
            "order": "Fabales",
            "family": "Fabaceae",
            "genus": "Glycine",
            "species": "max",
            "common_name": "Soybean",
            "authority": "(Linnaeus) Merrill",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "NC",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=GLMA4",
            "note": null,
            "hex": "#a0b837",
            "rgb": [
                160,
                184,
                55
            ],
            "alias": []
        },
        "AMHY": {
            "class_id": 18,
            "USDA_symbol": "AMHY",
            "EPPO": "AMACH",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Caryophyllidae",
            "order": "Caryophyllales",
            "family": "Amaranthaceae",
            "genus": "Amaranthus",
            "species": "hybridus",
            "common_name": "Smooth pigweed",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "MD",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=AMHY",
            "note": null,
            "hex": "#b360e8",
            "rgb": [
                179,
                96,
                232
            ],
            "alias": []
        },
        "CHAL7": {
            "class_id": 19,
            "USDA_symbol": "CHAL7",
            "EPPO": "CHEAL",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Caryophyllidae",
            "order": "Caryophyllales",
            "family": "Chenopodiaceae",
            "genus": "Chenopodium",
            "species": "album",
            "common_name": "Common lambsquarters",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "MD",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=CHAL7",
            "note": null,
            "hex": "#6ba33c",
            "rgb": [
                107,
                163,
                60
            ],
            "alias": []
        },
        "PADI": {
            "class_id": 20,
            "USDA_symbol": "PADI",
            "EPPO": "PANDI",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Panicum",
            "species": "dichotomiflorum",
            "common_name": "Fall panicum",
            "authority": "Michaux",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "MD",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=PADI",
            "note": null,
            "hex": "#ab2a9f",
            "rgb": [
                171,
                42,
                159
            ],
            "alias": []
        },
        "DAST": {
            "class_id": 21,
            "USDA_symbol": "DAST",
            "EPPO": "DATST",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Solanales",
            "family": "Solanaceae",
            "genus": "Datura",
            "species": "stramonium",
            "common_name": "Jimson weed",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb, subshrub",
            "duration": "annual",
            "collection_location": "MD",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=DAST",
            "note": null,
            "hex": "#cfea7c",
            "rgb": [
                207,
                234,
                124
            ],
            "alias": []
        },
        "ABTH": {
            "class_id": 22,
            "USDA_symbol": "ABTH",
            "EPPO": "ABUTH",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Dilleniidae",
            "order": "Malvales",
            "family": "Malvaceae",
            "genus": "Abutilon",
            "species": "theophrasti",
            "common_name": "Velvetleaf",
            "authority": "Medicus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "MD",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=ABTH",
            "note": null,
            "hex": "#8e42b0",
            "rgb": [
                142,
                66,
                176
            ],
            "alias": []
        },
        "SEPU8": {
            "class_id": 23,
            "USDA_symbol": "SEPU8",
            "EPPO": "SETPU",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Setaria",
            "species": "pumila",
            "common_name": "Yellow foxtail",
            "authority": "(C. Linnaeus) Romer and Schultes ",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "MD",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=SEPU8",
            "note": null,
            "hex": "#40be7b",
            "rgb": [
                64,
                190,
                123
            ],
            "alias": []
        },
        "SEFA": {
            "class_id": 24,
            "USDA_symbol": "SEFA",
            "EPPO": "SETFA",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Setaria",
            "species": "faberi",
            "common_name": "Giant foxtail",
            "authority": "Herrmann",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "MD",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=SEFA",
            "note": null,
            "hex": "#e636a0",
            "rgb": [
                230,
                54,
                160
            ],
            "alias": []
        },
        "ERCA20": {
            "class_id": 25,
            "USDA_symbol": "ERCA20",
            "EPPO": "ERICA",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Asterales",
            "family": "Asteraceae",
            "genus": "Erigeron",
            "species": "canadensis",
            "common_name": "Horseweed",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual biennial",
            "collection_location": "MD",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": null,
            "note": "Name change",
            "hex": "#5aedc3",
            "rgb": [
                90,
                237,
                195
            ],
            "alias": []
        },
        "ZEA": {
            "class_id": 26,
            "USDA_symbol": "ZEA",
            "EPPO": "ZEAMX",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Zea",
            "species": "mays",
            "common_name": "Maize",
            "authority": "Linnaeus",
            "growth_habit": "graminoid",
            "duration": "annual perenial",
            "collection_location": "MD",
            "category": "warm season weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=ZEA",
            "note": null,
            "hex": "#e83c2b",
            "rgb": [
                232,
                60,
                43
            ],
            "alias": [
                "corn"
            ]
        },
        "plant": {
            "class_id": 27,
            "USDA_symbol": "plant",
            "EPPO": "unknown",
            "group": "unknown",
            "class": "unknown",
            "subclass": "unknown",
            "order": "unknown",
            "family": "unknown",
            "genus": "unknown",
            "species": "unknown",
            "common_name": "unknown",
            "authority": "unknown",
            "growth_habit": "unknown",
            "duration": "unknown",
            "collection_location": "unknown",
            "category": "unknown",
            "collection_timing": "unknown",
            "multi_species_USDA_symbol": null,
            "link": null,
            "note": null,
            "hex": "#55ece3",
            "rgb": [
                85,
                236,
                227
            ],
            "alias": []
        },
        "colorchecker": {
            "class_id": 28,
            "USDA_symbol": "colorchecker",
            "EPPO": "colorchecker",
            "group": "colorchecker",
            "class": "colorchecker",
            "subclass": "colorchecker",
            "order": "colorchecker",
            "family": "colorchecker",
            "genus": "colorchecker",
            "species": "colorchecker",
            "common_name": "colorchecker",
            "authority": "colorchecker",
            "growth_habit": "colorchecker",
            "duration": "colorchecker",
            "collection_location": "colorchecker",
            "category": "colorchecker",
            "collection_timing": "colorchecker",
            "multi_species_USDA_symbol": null,
            "link": null,
            "note": null,
            "hex": "#e73b58",
            "rgb": [
                231,
                59,
                88
            ],
            "alias": []
        },
        "VIVI": {
            "class_id": 29,
            "USDA_symbol": "VIVI",
            "EPPO": "VICVI",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Rosidae",
            "order": "Fabales",
            "family": "Fabaceae",
            "genus": "Vicia",
            "species": "villosa",
            "common_name": "Hairy vetch",
            "authority": "Roth",
            "growth_habit": "forb/herb vine",
            "duration": "annual biennial perennial",
            "collection_location": "MD",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=VIVI",
            "note": null,
            "hex": "#9ee3a0",
            "rgb": [
                158,
                227,
                160
            ],
            "alias": []
        },
        "PISA6": {
            "class_id": 30,
            "USDA_symbol": "PISA6",
            "EPPO": "PIBSX",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Rosidae",
            "order": "Fabales",
            "family": "Fabaceae",
            "genus": "Pisum",
            "species": "sativum",
            "common_name": "Winter pea",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb vine",
            "duration": "annual",
            "collection_location": "MD",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=PISA6",
            "note": null,
            "hex": "#617ff7",
            "rgb": [
                97,
                127,
                247
            ],
            "alias": []
        },
        "TRIN3": {
            "class_id": 31,
            "USDA_symbol": "TRIN3",
            "EPPO": "TRFIN",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Rosidae",
            "order": "Fabales",
            "family": "Fabaceae",
            "genus": "Trifolium",
            "species": "incarnatum",
            "common_name": "Crimson clover",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "MD",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=TRIN3",
            "note": null,
            "hex": "#eea626",
            "rgb": [
                238,
                166,
                38
            ],
            "alias": []
        },
        "TRPR2": {
            "class_id": 32,
            "USDA_symbol": "TRPR2",
            "EPPO": "TRFPR",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Rosidae",
            "order": "Fabales",
            "family": "Fabaceae",
            "genus": "Trifolium",
            "species": "pratense",
            "common_name": "Red clover",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual biennial",
            "collection_location": "MD",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=TRPR2",
            "note": null,
            "hex": "#3967c5",
            "rgb": [
                57,
                103,
                197
            ],
            "alias": []
        },
        "Brassica_complex_0": {
            "class_id": 33,
            "USDA_symbol": "BRASS2",
            "EPPO": "1BRSG",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Dilleniidae",
            "order": "Capparales",
            "family": "Brassicaceae",
            "genus": "Brassica",
            "species": "spp.",
            "common_name": "Mustards",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual biennial perennial",
            "collection_location": "TX",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": [
                "BRNA",
                "BRRA",
                "BRJU",
                "SIAL5"
            ],
            "link": "https://plants.usda.gov/home/plantProfile?symbol=BRASS2",
            "note": "Includes Brassica napus (BRNA), Brassica rapa (BRRA), Brassica juncea (BRJU), Sinapis alba (SIAL5)",
            "hex": "#e86928",
            "rgb": [
                232,
                105,
                40
            ],
            "alias": [
                "brassica complex"
            ]
        },
        "RASA2": {
            "class_id": 34,
            "USDA_symbol": "RASA2",
            "EPPO": "RAPSR",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Dilleniidae",
            "order": "Capparales",
            "family": "Brassicaceae",
            "genus": "Raphanus",
            "species": "sativus",
            "common_name": "cultivated radish",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual biennial",
            "collection_location": "TX",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=RASA2",
            "note": null,
            "hex": "#5e97e6",
            "rgb": [
                94,
                151,
                230
            ],
            "alias": [
                "raphanus sativus"
            ]
        },
        "SECE": {
            "class_id": 35,
            "USDA_symbol": "SECE",
            "EPPO": "SECCE",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Secale",
            "species": "cereale",
            "common_name": "Cereal rye",
            "authority": "Linnaeus",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "NC, MD",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=SECE",
            "note": null,
            "hex": "#ba3a1c",
            "rgb": [
                186,
                58,
                28
            ],
            "alias": []
        },
        "TRITI2": {
            "class_id": 36,
            "USDA_symbol": "TRITI2",
            "EPPO": "1TTLG",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Triticosecale",
            "species": "Triticosecale",
            "common_name": "Triticale",
            "authority": "Wittm. ex A. Camus",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "NC",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=TRITI2",
            "note": null,
            "hex": "#5dcfe9",
            "rgb": [
                93,
                207,
                233
            ],
            "alias": []
        },
        "TRAE": {
            "class_id": 37,
            "USDA_symbol": "TRAE",
            "EPPO": "TRZAX",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Triticum",
            "species": "aestivum",
            "common_name": "Winter wheat",
            "authority": "Linnaeus",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "NC",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=TRAE",
            "note": null,
            "hex": "#ea427e",
            "rgb": [
                234,
                66,
                126
            ],
            "alias": []
        },
        "AVSA": {
            "class_id": 38,
            "USDA_symbol": "AVSA",
            "EPPO": "AVESA",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Avena",
            "species": "sativa",
            "common_name": "Oats",
            "authority": "Linnaeus",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "NC, TX",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=AVSA",
            "note": null,
            "hex": "#44975e",
            "rgb": [
                68,
                151,
                94
            ],
            "alias": []
        },
        "HORDE": {
            "class_id": 39,
            "USDA_symbol": "HORDE",
            "EPPO": "1HORG",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Hordeum",
            "species": "Hordeum",
            "common_name": "Barley",
            "authority": "Linnaeus",
            "growth_habit": "graminoid",
            "duration": "annual perenial",
            "collection_location": "NC",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=HORDE",
            "note": null,
            "hex": "#e172d8",
            "rgb": [
                225,
                114,
                216
            ],
            "alias": []
        },
        "AVST2": {
            "class_id": 40,
            "USDA_symbol": "AVST2",
            "EPPO": "AVESG",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Avena",
            "species": "strigosa",
            "common_name": "Black oats",
            "authority": "Schreber",
            "growth_habit": "graminoid",
            "duration": "annual",
            "collection_location": "NC, TX",
            "category": "cool season cover crop",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=AVST2",
            "note": null,
            "hex": "#407018",
            "rgb": [
                64,
                112,
                24
            ],
            "alias": []
        },
        "GOHI": {
            "class_id": 41,
            "USDA_symbol": "GOHI",
            "EPPO": "GOSHI",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Dilleniidae",
            "order": "Malvales",
            "family": "Malvaceae",
            "genus": "Gossypium",
            "species": "hirsutum",
            "common_name": "upland cotton",
            "authority": " Linnaeus",
            "growth_habit": "Shrub",
            "duration": "annual",
            "collection_location": "TX",
            "category": "summer cash crop",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=GOHI",
            "note": null,
            "hex": "#bd7ae4",
            "rgb": [
                189,
                122,
                228
            ],
            "alias": [
                "cotton"
            ]
        },
        "URRE2": {
            "class_id": 42,
            "USDA_symbol": "URRE2",
            "EPPO": "PANRP",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Urochloa",
            "species": "reptans",
            "common_name": "Sprawling signalgrass",
            "authority": "(Linnaeus) Stapf",
            "growth_habit": "Graminoid",
            "duration": "annual perennial",
            "collection_location": "TX",
            "category": "summer weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=URRE2",
            "note": null,
            "hex": "#aa952b",
            "rgb": [
                170,
                149,
                43
            ],
            "alias": []
        },
        "SISP": {
            "class_id": 43,
            "USDA_symbol": "SISP",
            "EPPO": " SIDSP",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Dilleniidae",
            "order": "Malvales",
            "family": "Malvaceae",
            "genus": "Sida",
            "species": "spinosa",
            "common_name": "Prickly sida",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb, subshrub",
            "duration": "annual perennial",
            "collection_location": "MD",
            "category": "summer weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=SISP",
            "note": null,
            "hex": "#7350af",
            "rgb": [
                115,
                80,
                175
            ],
            "alias": []
        },
        "IPLA": {
            "class_id": 44,
            "USDA_symbol": "IPLA",
            "EPPO": "IPOLA",
            "group": "Dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Solanales",
            "family": "Convolvulaceae",
            "genus": "Ipomoea",
            "species": "lacunosa",
            "common_name": "Pitted morning-glory",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb vine",
            "duration": "annual",
            "collection_location": "MD",
            "category": "summer weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=IPLA",
            "note": null,
            "hex": "#d6a846",
            "rgb": [
                214,
                168,
                70
            ],
            "alias": []
        },
        "AMSP": {
            "class_id": 45,
            "USDA_symbol": "AMSP",
            "EPPO": "AMASP",
            "group": "Dicot",
            "class": "Magnoliopsida",
            "subclass": "Caryophyllidae",
            "order": "Caryophyllales",
            "family": "Amaranthaceae",
            "genus": "Amaranthus",
            "species": "spinosus",
            "common_name": "Spiny amaranth",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": "MD",
            "category": "summer weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=AMSP",
            "note": null,
            "hex": "#968de9",
            "rgb": [
                150,
                141,
                233
            ],
            "alias": []
        },
        "TRPO2": {
            "class_id": 46,
            "USDA_symbol": "TRPO2",
            "EPPO": "TRTPO",
            "group": "Dicot",
            "class": "Magnoliopsida",
            "subclass": "Caryophyllidae",
            "order": "Caryophyllales",
            "family": "Aizoaceae",
            "genus": "Trianthema",
            "species": "portulacastrum",
            "common_name": "desert horsepurslane",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual/perennial",
            "collection_location": "TX",
            "category": "summer weed",
            "collection_timing": "summer",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=TRPO2",
            "note": null,
            "hex": "#d88633",
            "rgb": [
                216,
                134,
                51
            ],
            "alias": []
        },
        "LOPEM2": {
            "class_id": 47,
            "USDA_symbol": "LOPEM2",
            "EPPO": " LOLMU",
            "group": "Monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Poaceae",
            "genus": "Lolium",
            "species": "perenne L. ssp. multiflorum",
            "common_name": "italian ryegrass",
            "authority": "(Lam) Husnot",
            "growth_habit": "graminoid",
            "duration": "annual/perennial",
            "collection_location": "NC",
            "category": "winter annual",
            "collection_timing": "winter",
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/home/plantProfile?symbol=LOPEM2",
            "note": null,
            "hex": "#34649f",
            "rgb": [
                52,
                100,
                159
            ],
            "alias": [
                "ryegrass"
            ]
        },
        "ACOS": {
            "class_id": 48,
            "USDA_symbol": "ACOS",
            "EPPO": "ACCOS",
            "group": "Dicot",
            "class": "Magnoliopsida",
            "subclass": "Rosidae",
            "order": "Euphorbiales",
            "family": "Euporbiaceae",
            "genus": "Acalypha",
            "species": "ostryifolia",
            "common_name": "pineland threeseed mercury",
            "authority": "Riddel",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": null,
            "category": "summer weed",
            "collection_timing": null,
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/plant-profile/ACOS",
            "note": null,
            "hex": "#34649f",
            "rgb": [
                52,
                100,
                159
            ],
            "alias": [
                "hophornbeam copperleaf"
            ]
        },
        "CYES": {
            "class_id": 49,
            "USDA_symbol": "CYES",
            "EPPO": "CYPES",
            "group": "monocot",
            "class": "Liliopsida",
            "subclass": "Commelinidae",
            "order": "Cyperales",
            "family": "Cyperaceae",
            "genus": "Cyperus",
            "species": "esculentus",
            "common_name": "yellow nutsedge",
            "authority": "Linnaeus",
            "growth_habit": "Graminoid",
            "duration": "perennial",
            "collection_location": null,
            "category": "summer weed",
            "collection_timing": null,
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/plant-profile/CYES",
            "note": null,
            "hex": "#e7d37e",
            "rgb": [
                231,
                211,
                126
            ],
            "alias": []
        },
        "AMTR": {
            "class_id": 50,
            "USDA_symbol": "AMTR",
            "EPPO": "AMBTR",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Asterales",
            "family": "Asteraceae",
            "genus": "Ambrosia",
            "species": "trfida",
            "common_name": "great ragweed",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb, subshrub",
            "duration": "annual",
            "collection_location": null,
            "category": "summer weed",
            "collection_timing": null,
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/plant-profile/AMTR",
            "note": null,
            "hex": "#9d4999",
            "rgb": [
                157,
                73,
                153
            ],
            "alias": [
                "giant ragweed"
            ]
        },
        "SOEL": {
            "class_id": 51,
            "USDA_symbol": "SOEL",
            "EPPO": "SOLEL",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Asterales",
            "family": "Solanaceae",
            "genus": "Solanum",
            "species": "elaegnifolium",
            "common_name": "silverleaf nightshade",
            "authority": "Cavaco",
            "growth_habit": "forb/herb, subshrub",
            "duration": "perennial",
            "collection_location": null,
            "category": "summer weed",
            "collection_timing": null,
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/plant-profile/SOEL",
            "note": null,
            "hex": "#7b852d",
            "rgb": [
                123,
                133,
                45
            ],
            "alias": []
        },
        "COAR4": {
            "class_id": 52,
            "USDA_symbol": "COAR4",
            "EPPO": "CONAR",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Solanales",
            "family": "Convolvulaceae",
            "genus": "Convolvulus",
            "species": "arvensis",
            "common_name": "field bindweed",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb, vine",
            "duration": "perennial",
            "collection_location": null,
            "category": "summer weed",
            "collection_timing": null,
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/plant-profile/COAR4",
            "note": null,
            "hex": "#b5317f",
            "rgb": [
                181,
                49,
                127
            ],
            "alias": []
        },
        "CIAR4": {
            "class_id": 53,
            "USDA_symbol": "CIAR4",
            "EPPO": "CIRAR",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Asterales",
            "family": "Asteraceae",
            "genus": "Cirsiium",
            "species": "arvensis",
            "common_name": "Canada thistle",
            "authority": "Linnaeus, Scopoli",
            "growth_habit": "forb/herb",
            "duration": "perennial",
            "collection_location": null,
            "category": "summer weed",
            "collection_timing": null,
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/plant-profile/CIAR4",
            "note": null,
            "hex": "#b5317f",
            "rgb": [
                181,
                49,
                127
            ],
            "alias": []
        },
        "AMRE": {
            "class_id": 54,
            "USDA_symbol": "AMRE",
            "EPPO": "AMARE",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Caryophyllidae",
            "order": "Caryophyllales",
            "family": "Amaranthaceae",
            "genus": "Amaranthus",
            "species": "retroflexus",
            "common_name": "redroot amaranth",
            "authority": "Linnaeus",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": null,
            "category": "summer weed",
            "collection_timing": null,
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/plant-profile/AMRE",
            "note": null,
            "hex": "#b5317f",
            "rgb": [
                181,
                49,
                127
            ],
            "alias": [
                "redroot pigweed"
            ]
        },
        "PRLO": {
            "class_id": 55,
            "USDA_symbol": "PRLO",
            "EPPO": "PROLU",
            "group": "dicot",
            "class": "Magnoliopsida",
            "subclass": "Asteridae",
            "order": "Scrophulariales",
            "family": "Pedaliaceae",
            "genus": "Proboscidea",
            "species": "louisianica",
            "common_name": "ram's horn",
            "authority": "Miller, Thellung",
            "growth_habit": "forb/herb",
            "duration": "annual",
            "collection_location": null,
            "category": "summer weed",
            "collection_timing": null,
            "multi_species_USDA_symbol": null,
            "link": "https://plants.usda.gov/plant-profile/PRLO",
            "note": null,
            "hex": "#b5317f",
            "rgb": [
                181,
                49,
                127
            ],
            "alias": [
                "devil's claw"
            ]
        }
    },
    "Notes": {
        "1": "USDA symbols are used as species index key.",
        "2": "When difference between USDA and EPPO codes, scientific_names, or authority occured, USDA information was used.",
        "3": "Purple nutsedge (CYRO) is used but may also be yellow nutsedge (CYES) for years 2022, and 2023.",
        "4": "North Carolina (MD), Texas (TX), Maryland (MD)",
        "5": "Brassicas class contains 4 species - Brassica napus, Brassica rapa, Brassica juncea, Brassica hirta",
        "6": "Large crabgrass collected in NC may be southern crabgrass.",
        "7": "Large crabgrass collected at any location maybe Large or other Spp.",
        "8": "Smooth pigweed (AMHY) may also be Redroot pigweed (AMRE).",
        "9": "USDA plant profile links for class id 48-55 are updated. The rest are not working."
    },
    "Version": 1.3
}
```

mkutu committed:
Update species_info.json

use empty lists instead of null. We plan to iterate through alias and can avoid any additional null checks. An empty list ([]) is preferable. It keeps data structures consistent and avoids potential null-related errors.

navjot-nangia committed:
corrected names for the cutout files

navjot-nangia committed:
added the inspection funcionality with inspection.py

navjot-nangia committed:
updated config files for inspection script

navjot-nangia committed:
updated inspection.py to avoid hardcoding, cleaned a bit

navjot-nangia committed:
Added detection confidence score to weed detection results and Batch_id

navjot-nangia committed:
cleaned code for det_pred_conf

navjot-nangia committed:
Merge branch 'incorporate_unet' into add_inspection_script

navjot-nangia committed:
changed the inspection reuslt layout to tight, added error info in loggin for unet script

mkutu committed:
Merge pull request #73 from precision-sustainable-ag/add_inspection_script

added the inspection functionality with inspection.py
